vulnerable.cpp

```
@@ -1 +1,108 @@
+        int *doubleFreePtr = new int(100);
+        delete doubleFreePtr;
+        delete doubleFreePtr;
+
+        // 根據輸入內容來觸發無終止條件的遞迴呼叫
+        if (inputCommand[0] == 'R') {
+            recursiveFunction();
+        }
+
+        // 空指標解引用（直接使用 NULL 指標）
+        int *nullPtr = nullptr;
+        cout << "Null pointer test: " << *nullPtr << endl;
```

Cppcheck (2.10-2)

[error] Null pointer dereference

(nullPointer)